camelot/view/action_steps/application.py

```diff
@@ -74,21 +74,23 @@ def render( self, gui_context ):
         app = QtWidgets.QApplication.instance()
         for widget in app.allWidgets():
             if isinstance(widget, QtWidgets.QMainWindow):
-                window = widget
-                break
+                # Make sure a QMainWindow is reused only once
+                if not hasattr(widget, '_reused_by_view_action_steps_application'):
```

Comment:
ik twijfel alleen aan deze lijn, ik heb niet graag 'hasattr' in de code, want dat kan rare side effects hebben

ook zit die setattr dan op de 'python' wrapper, terwijl het eigenlijk een C++ object is.

ik zie nu ook nt direct een andere methode, mss checken op het bestaan van één of andere child, ofwel het object hier een objectName geven en die dan checken.